client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx

```diff
@@ -446,23 +454,33 @@ const UnifiedDesignPickerStep: Step = ( { navigation, flow, stepName } ) => {
 		} else {
 			plan = isEligibleForProPlan && isEnabled( 'plans/pro-plan' ) ? 'pro' : 'premium';
 		}
+		goToCheckout( {
+			flowName: flow,
+			stepName,
+			siteSlug: siteSlug || urlToSlug( site?.URL || '' ) || '',
+			// When the user is done with checkout, send them back to the current url
+			destination: window.location.href.replace( window.location.origin, '' ),
+			plan,
+			extraProducts:
+				selectedDesign?.is_externally_managed && isMarketplaceThemeSubscriptionNeeded
+					? [ marketplaceProductSlug ]
+					: [],
+		} );
+	}
+	function handleCheckout() {
+		recordTracksEvent( 'calypso_signup_design_upgrade_modal_checkout_button_click', {
+			theme: selectedDesign?.slug,
+			is_externally_managed: selectedDesign?.is_externally_managed,
+		} );
 
 		if ( siteSlugOrId ) {
-			goToCheckout( {
-				flowName: flow,
-				stepName,
-				siteSlug: siteSlug || urlToSlug( site?.URL || '' ) || '',
-				// When the user is done with checkout, send them back to the current url
-				destination: window.location.href.replace( window.location.origin, '' ),
-				plan,
-				extraProducts:
-					selectedDesign?.is_externally_managed && isMarketplaceThemeSubscriptionNeeded
-						? [ marketplaceProductSlug ]
-						: [],
-			} );
-
-			setShowUpgradeModal( false );
+			if ( selectedDesign?.is_externally_managed && hasEligibilityMessages ) {
```

Comment:
I would put an event here to track how many clients get to this step, it might help us to define the funnels too.

Reply:
Great idea, I would create a follow-up task for that

Reply:
Dropping the created issue here https://github.com/Automattic/wp-calypso/issues/81768